lib/std/event/loop.zig

```diff
@@ -512,21 +512,28 @@ pub const Loop = struct {
             },
             .kev = undefined,
         };
-        defer self.bsdRemoveKev(ident, filter);
+
+        defer {
+            // If the kevent was set to be ONESHOT, it doesn't need to be deleted manually.
+            if (flags & os.EV_ONESHOT != 0) {
+                self.bsdRemoveKev(ident, filter);
+            }
+        }
+
         suspend {
-            self.bsdAddKev(&resume_node, ident, filter, fflags) catch unreachable;
+            self.bsdAddKev(&resume_node, ident, filter, flags) catch unreachable;
         }
     }
 
     /// resume_node must live longer than the anyframe that it holds a reference to.
-    pub fn bsdAddKev(self: *Loop, resume_node: *ResumeNode.Basic, ident: usize, filter: i16, fflags: u32) !void {
+    pub fn bsdAddKev(self: *Loop, resume_node: *ResumeNode.Basic, ident: usize, filter: i16, flags: u16) !void {
```

Comment:
Wow nice catch on the filter flags, I can't believe I was passing EV_ONESHOT to the wrong parameter :cold_sweat: